common/djangoapps/entitlements/tests/test_entitlements_api.py

```diff
@@ -0,0 +1,162 @@
+    ENABLED_CACHES = ['default']
+    COURSE_RUN_ID = 'some/great/course'
+
+    def setUp(self):
+        super(EntitlementsTest, self).setUp()
+        self.user = UserFactory(is_staff=True)
+        self.client.login(username=self.user.username, password=TEST_PASSWORD)
+        self.course = CourseFactory.create(org='TestX', course='TS101', run='T1')
+        # self.entitlements_url = '/api/entitlements/v1/entitlements/'
+        self.entitlements_url = reverse('entitlements_api:entitlements-list')
+        self.entitlements_uuid_path = 'entitlements_api:api:entitlements-list'
+        self.course_uuid = str(uuid.uuid4())
+
+    def _get_data_set(self, user):
+        """
+        Get a basic data set for an entitlement
+        """
+        return {
+            "user": user.username,
+            "mode": "verified",
+            "course_uuid": self.course_uuid,
+            "order_number": "EDX-1001"
+        }
+
+    def test_auth_required(self):
+        self.client.logout()
+        response = self.client.get(self.entitlements_url)
+        assert response.status_code == 401
+
+    def test_staff_user_required(self):
+        not_staff_user = UserFactory()
+        self.client.login(username=not_staff_user.username, password=UserFactory._DEFAULT_PASSWORD)
+        response = self.client.get(self.entitlements_url)
+        assert response.status_code == 403
+
+    def test_add_entitlement_with_missing_data(self):
+        entitlement_data_missing_parts = self._get_data_set(self.user)
+        entitlement_data_missing_parts.pop('mode')
+        entitlement_data_missing_parts.pop('course_uuid')
+
+        response = self.client.post(
+            self.entitlements_url,
+            data=json.dumps(entitlement_data_missing_parts),
+            content_type='application/json',
+        )
+        assert response.status_code == 400
+
+    def test_add_entitlement(self):
+        entitlement_data = self._get_data_set(self.user)
+
+        response = self.client.post(
+            self.entitlements_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+        assert response.status_code == 201
+        results = response.data
+        assert results is not None
+
+        course_entitlement = CourseEntitlement.objects.get(
+            user=self.user,
+            course_uuid=self.course_uuid
+        )
+
+        self.assertIsNotNone(course_entitlement)
+        self.assertEqual(results['uuid'], str(course_entitlement.uuid))
+        self.assertEqual(str(course_entitlement.course_uuid), self.course_uuid)
+        self.assertEqual(course_entitlement.enrollment_course_run, None)
+        self.assertEqual(course_entitlement.mode, 'verified')
+        self.assertIsNotNone(course_entitlement.created)
+        self.assertTrue(course_entitlement.user, self.user)
+
+    def test_get_entitlements(self):
+        CourseEntitlementFactory.create(user=self.user, order_number='TESTX-1001')
```

Comment:
No need for `create`. Pass the kwargs on `CourseEntitlementFactory()`. Also, if you update the factory to generate order numbers, you can ditch those kwargs.

Reply:
If the user doesn't matter for your tests, just call `CourseEntitlementFactory.create_batch(2)`